spec/enumerable_spec.rb

```diff
@@ -0,0 +1,209 @@
+    end
+
+    it 'accepts a hash' do
+      expect(hash.my_each_with_index { |num, _idx| num }).to eql(hash)
+    end
+
+    it 'returns Enumerator when block is not given' do
+      expect(array.my_each_with_index).to be_an(Enumerator)
+    end
+  end
+
+  describe '#my_select' do
+    it 'returns Enumerator when block is not given' do
+      expect(array.my_select).to be_an(Enumerator)
+    end
+
+    it 'returns a filter array' do
+      expect(array.my_select { |num| num > 3 }).to eql([4, 5])
+    end
+
+    it 'returns a filter array for a range' do
+      expect(range.my_select { |num| num < 3 }).to eql([0, 1, 2])
+    end
+
+    it 'returns a filter array for a hash' do
+      expect(hash.my_select { |_k, v| v > 2 }).to eql({ c: 3 })
+    end
+  end
+
+  describe '#my_all' do
+    it 'returns true when all values are true on array' do
+      expect(array.my_all? { |n| n > 2 }).to eql FALSE
+    end
+
+    it 'returns false when one values is false on array' do
+      expect(array.my_all? { |n| n > 6 }).to eql FALSE
+    end
+
+    it 'returns false when one values is false on range' do
+      expect(range.my_all? { |n| n > 2 }).to eql FALSE
+    end
+
+    it 'returns true when all values are true on range' do
+      expect(range.my_all? { |n| n < 6 }).to eql TRUE
+    end
+
+    it 'true when all values are true on hash' do
+      expect(hash.my_all? { |_k, v| v < 7 }).to eql TRUE
+    end
+
+    it 'false when one values is false on range' do
+      expect(hash.my_all? { |_k, v| v > 3 }).to eql FALSE
+    end
+  end
+
+  describe '#my_any?' do
+    it 'true when any value is true on the array' do
+      expect(array.my_any? { |n| n > 2 }).to eql TRUE
+    end
+
+    it 'false when no value is true on the array' do
+      expect(array.my_any? { |n| n > 6 }).to eql FALSE
+    end
+
+    it 'true when any value is true on the range' do
+      expect(range.my_any? { |n| n > 2 }).to eql TRUE
+    end
+
+    it 'false when no value is true on the range' do
+      expect(range.my_any? { |n| n > 6 }).to eql FALSE
+    end
+
+    it 'true when any value is true on the hash' do
+      expect(hash.my_any? { |_k, v| v > 2 }).to eql TRUE
+    end
+
+    it 'false when no value is true on the hash' do
+      expect(hash.my_any? { |_k, v| v > 6 }).to eql FALSE
+    end
+  end
+
+  describe '#my_none?' do
+    it 'false when no value is true on the array' do
+      expect(array.my_none? { |n| n > 2 }).to eql FALSE
+    end
+
+    it 'true when any value is true on the array' do
+      expect(array.my_none? { |n| n > 6 }).to eql TRUE
+    end
+
+    it 'false when no value is true on the range' do
+      expect(range.my_none? { |n| n > 2 }).to eql FALSE
+    end
+
+    it 'false when any value is true on the range' do
+      expect(range.my_none? { |n| n > 6 }).to eql TRUE
+    end
+
+    it 'false when no value is true on the hash' do
+      expect(hash.my_none? { |_k, v| v > 2 }).to eql FALSE
+    end
```

Comment:
This statement should say, it returns `true` when no value is true on the hash. Yours says `false`.